openedx/core/djangoapps/content/block_structure/signals.py

```diff
@@ -14,7 +14,7 @@ def _listen_for_course_publish(sender, course_key, **kwargs):  # pylint: disable
     Catches the signal that a course has been published in the module
     store and creates/updates the corresponding cache entry.
     """
-    update_course_in_cache.apply_async([unicode(course_key)], {'desired_queue_env': 'lms'}, countdown=0)
+    update_course_in_cache.apply_async([unicode(course_key)], countdown=0)
```

Comment:
I had to move from using this kwargs approach to defining `ALTERNATE_ENV_TASKS` in `celery.py`, because using kwargs this way would break on devstack - with CELERY_ALWAYS_EAGER set to True, `route_for_task` was never being called and `desired_queue_env` would get passed through to a function that wasn't expecting it.

I think the only way we're going to be able to use routing is to define "special" tasks in some centralized location, away from where they're declared.